apps/src/music/blockly/advancedProcedures.ts

```diff
@@ -0,0 +1,9 @@
+import musicI18n from '../locale';
+
+export function installAdvancedProcedures() {
+  Blockly.Extensions.unregister('procedure_caller_mutator');
+  Blockly.Extensions.unregister('procedure_def_mutator');
+  require('@blockly/block-plus-minus');
```

Comment:
Is it possible/desirable to use `import` here?

Reply:
Unfortunately, not.

Using `import` instead of `require` causes the errors like `"Extension "procedure_def_mutator" is already registered"` because the import is executed when the module is initially loaded. This means that it begins registering blocks and mutators as soon as the script is loaded, even before we know if we should be in advanced mode. I also considered in async function containing the import, but that caused the opposite problem.

Reply:
Sounds like `import()` and handling the returned promise would be possible, but complex!